trie-db/src/lookup.rs

```diff
@@ -655,4 +668,175 @@ where
 		}
 		Ok(None)
 	}
+
+	/// Look up the merkle value (hash) of the node that is the closest descendant for the provided
+	/// key.
+	///
+	/// When the provided key leads to a node, then the merkle value of that node
+	/// is returned. However, if the key does not lead to a node, then the merkle value
+	/// of the closest descendant is returned. `None` if no such descendant exists.
+	fn lookup_first_descendent_without_cache(
+		mut self,
+		nibble_key: NibbleSlice,
+		full_key: &[u8],
+	) -> Result<Option<TrieHash<L>>, TrieHash<L>, CError<L>> {
+		let mut partial = nibble_key;
+		let mut hash = self.hash;
+		let mut key_nibbles = 0;
+
+		// this loop iterates through non-inline nodes.
+		for depth in 0.. {
+			let node_data = match self.db.get(&hash, nibble_key.mid(key_nibbles).left()) {
```

Comment:
That makes sense! Have written the code to support both pathways:
- I've used the cache if it exits
- If we don't have any caching enabled for the query; then I've transformed the decoded node into `NodeOwned`

Since we are now working with a partial NibbleSlice and a NodeOwned::NibbleVec, I've also added a `starts_with_slice` method (since the NibbleVec could have a padding) 

Let me know what you think of it, thanks! 🙏